commonmark/src/main/java/org/commonmark/renderer/html/CoreHtmlNodeRenderer.java

```diff
@@ -274,6 +276,44 @@ private void renderListBlock(ListBlock listBlock, String tagName, Map<String, St
         html.line();
     }
 
+    private static MediaType detectMediaType(String url) {
+        int extensionStart = url.lastIndexOf('.');
```

Comment:
This won't work if the URL happens to have query params that include a dot, for example `/url.mp4?param=foo.bar`. We probably need to parse the URL here and get the extension of the path that way.